lib/srv/ctx.go

```diff
@@ -695,7 +694,7 @@ func (c *ServerContext) OpenXServerListener(x11Req x11.ForwardRequestPayload, di
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	originPortI, err := strconv.Atoi(originPort)
+	originPortI, err := apiutils.StrToUInt16(originPort)
```

Comment:
Similarly here, there's no way this check ever triggers an error since the the host port is coming from an existing conn (meaning the port must be valid). All we've done is added more branches and indirection and not increased security in any meaningful way.